packages/flame/test/components/component_test.dart

```diff
@@ -28,6 +28,91 @@ void main() {
         await expectLater(component.mounted, completes);
       });
 
+      testWithFlameGame(
+          'component removed times out when set before game is ready',
```

Comment:
This shouldn't be the expected behavior though, right? If such future is awaited, the game will enter into an infinite loop...

Reply:
What do you suggest should be returned? Returning a completed feature is also wrong, since the component haven't been removed yet. Why will it enter an infinite loop?

This test seems wrong though, because here the component is actually removed and then it still times out?
And the description of the test should be changed, once we figure out what the intended behavior should be.

Reply:
I would expect the future to complete upon the component's removal even if it was created before the game was mounted.

If for some reason that's impossible, then it would be better to throw an `assert` error at the time of the future creation, if it can be detected that this future would not operate properly.

Reply:
Agree, I think the test is just wrong. I wonder why it times out though.

Reply:
@spydon i found why the `expectLater(removed, completes)` just times out. When trying to run `_finishRemoving` that I added to the `propagateToChildren` call that I added in `_remove()`. The `_lifecycleManager` is null so it never arrives at the completer for `removed` on the component. 

However, not sure why the `_lifecycleManager` is null in this stage.

Reply:
I pushed a change that should keep the `_lifecyclemanager` alive until its `_removedCompleter` has run `.complete()` and been set to null. Let me know what you think.